test/integration/core/ledger/transaction.spec.ts

```diff
@@ -329,6 +330,51 @@ describe("on us payment", () => {
     await expectBalance({ account: payee.accountPath, currency: "BTC", balance: 0 })
     await expectBalance({ account: payee.accountPath, currency: "USD", balance: 0.1 })
   })
+  it("does not share memo when below sats threshold", async () => {
+    const payer = walletBTC
+    const payee = walletBTC2
+    const spamMsg = "THIS MIGHT BE SPAM!!!"
+
+    const entry = await ledger.addOnUsPayment({
+      description: "desc",
+      sats: MEMO_SHARING_SATS_THRESHOLD - 1,
+      metadata: { type: "on_us", pending: false },
+      payerUser: payer,
+      payeeUser: payee,
+      lastPrice: UserWallet.lastPrice,
+      memoFromPayer: spamMsg,
+      shareMemoWithPayee: true,
+    })
+
+    for (const txn of entry.transactions) {
+      if (txn.credit > 0) {
+        expect(txn.memoFromPayer).toBeUndefined()
+      }
+    }
+  })
+  it("shares memo when above or equal to sats threshold", async () => {
+    const payer = walletBTC
+    const payee = walletBTC2
+    const spamMsg = "THIS MIGHT BE SPAM!!!"
+
+    const entry = await ledger.addOnUsPayment({
+      description: "desc",
+      sats: MEMO_SHARING_SATS_THRESHOLD,
+      metadata: { type: "on_us", pending: false },
+      payerUser: payer,
+      payeeUser: payee,
+      lastPrice: UserWallet.lastPrice,
+      memoFromPayer: spamMsg,
+      shareMemoWithPayee: true,
+    })
+
+    for (const txn of entry.transactions) {
+      console.log(txn)
+      if (txn.credit > 0) {
```

Comment:
This seems like an implicit way to identify this?

Reply:
I think @samerbuna and I were assuming that if a transaction has a memo and a non-null `txn.credit` value then it is a "from someone" transaction and at that point the test checks that the expected `memoFromPayer` value is received based on the `sats` value originally passed into `addInvoice`.